### ISSUE TITLE
No-debug mode

### ISSUE THREAD

**Issue by polinasok** [go-delve/delve#2318]
This request is driven by a `Run Without Debugging` feature in vscode and `noDebug` launch flag in [DAP](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Launch). It supported by the original vscode-go adapter that actually uses dlv to "run" programs in some cases. Please see https://github.com/golang/vscode-go/issues/1111 for details.

I can imagine that this can be relevant outside of DAP/vscode. If a user sets up a configuration (or even just a command) with various flags and options, it can be convenient to reuse it as-is with and without debugging and not have to copy all the settings in two places. As is explained in https://github.com/golang/vscode-go/issues/1111, "no debug" doesn't mean "no debugger", but to "run the program without hitting breakpoints and breaking on exceptions." 

Dlv has two hidden breakpoints. Looks like they are not subject to `clearall`, but can be cleared one-by-one.
```
$ dlv debug helloworld.go 
Type 'help' for list of commands.
(dlv) breakpoints
Breakpoint runtime-fatal-throw at 0x1033d10 for runtime.fatalthrow() /usr/local/go/src/runtime/panic.go:1162 (0)
Breakpoint unrecovered-panic at 0x1033d80 for runtime.fatalpanic() /usr/local/go/src/runtime/panic.go:1189 (0)
	print runtime.curg._panic.arg
(dlv) clearall
(dlv) breakpoints
Breakpoint runtime-fatal-throw at 0x1033d10 for runtime.fatalthrow() /usr/local/go/src/runtime/panic.go:1162 (0)
Breakpoint unrecovered-panic at 0x1033d80 for runtime.fatalpanic() /usr/local/go/src/runtime/panic.go:1189 (0)
	print runtime.curg._panic.arg
(dlv) clear unrecovered-panic
Breakpoint unrecovered-panic cleared at 0x1033d80 for runtime.fatalpanic() /usr/local/go/src/runtime/panic.go:1189
(dlv) breakpoints
Breakpoint runtime-fatal-throw at 0x1033d10 for runtime.fatalthrow() /usr/local/go/src/runtime/panic.go:1162 (0)
(dlv) 
```
Is that what the clients should do or should this be added as a command-line option?
Is there anything else that should be streamlined for a no-debugging experience?

**Comment by aarzilli** [comment]
Even disabling those breakpoints the debugger will still stop on hardcoded breakpoints (`runtime.Breakpoint()` calls). A true noDebug experience should probably be implemented with `exec.Command(...)` if I understand correctly.

**Comment by polinasok** [comment]
If there is `runtime.Breakpoint()`, `go run` will hit SIGTRAP and not run to completion like there are no breakpoints either. So I would not worry about hardcoded breakpoints. I think it is fair to implement no-debug as no-debugging, not necessarily no-debugger, unless that's just easier.

**Comment by aarzilli** [comment]
I guess it's nitpick but a program running outside a debugger that hits a hardcoded breakpoint will exit, printing the SIGTRAP message, but if it runs inside a debugger it will stop in the debugger.

**Comment by polinasok** [comment]
A version of noDebug for dap mode using `exec.Command` has been added in #2400.

**Comment by aarzilli** [comment]
Closing, implemented in 3c1b94276a31d7992465637223d5f92941941b2b

**Comment by aarzilli** [comment]
I'm not a fan of this in general. To me it doesn't seem very useful, no other debugger has it and we didn't have user requests for it. I wish DAP didn't have this feature either, nodebug modes don't belong in debuggers IMO. DAP has it and it's fine to implement it, but I wouldn't spread it around. The `trace` subcommand is an approximation anyway.
